### PR TITLE
Add Central Controller + Events

### PR DESCRIPTION
**Pull request by AidanCampbell738**
This PR:
- adds the central controller to the python model. This controller prompts users for events and contains a help command to list all possible events.
- splits transitions into two functions: a guard function that checks if a transition can be taken and another that actually executes the transitions 
- implements transition onExpr (in transition guard function)
- implements transition sendExpr (return value of transition execution function)
- adds a handleExpression function to the state class
- adds a dictionary to each state that maps guard transition functions to transition execution functions (which is used in handleExpression)

See linked Asana tickets for a more in-depth description and sample input/output

TODO:
- is there a need to differentiate between env events and regular events? (maybe ask Prof Day)
- add relevant test cases to test files

Ticket links:
https://app.asana.com/0/1200958948599566/1202460837475461/f
https://app.asana.com/0/1200958948599566/1202309691137488/f

### CODE REVIEW

File: org.alloytools.alloy.application/src/main/java/templates/TestTemplateFile.vm [hunk 1]
```diff
@@ -105,17 +120,16 @@ ${signature.Name} = Signature("${signature.Name}", "${signature.Multiplicity}",
...
 		#set($tab = "    ")
 		#set($newpadding = "$padding$tab")
 		#generateState($substate $newpadding)
 	#end
 
-$padding    """ The wrapper state for ${concState.getName()} """
-
 $padding    def __init__(self, context: State, root_context: State) -> None:
-$padding        super(${concState.getName()}, self).__init__(self) # init superclass
+$padding        super({concState.getName()}_State, self).__init__(self) # init superclass
```
Comment (Chestshire) [inline org.alloytools.alloy.application/src/main/java/templates/TestTemplateFile.vm]: Missed `$` in front of `{concState.getName()}`

Comment (AidanCampbell738) [reply org.alloytools.alloy.application/src/main/java/templates/TestTemplateFile.vm]: Good catch. Fixed.

**Comment by Simonliuwaterloo** [comment]
image
I am getting a "Inconsistent use of tabs and spaces in indentationPylance" when translating Bitcounter on line 332.

**Comment by Chestshire** [comment]
> image
> 
> I am getting a "Inconsistent use of tabs and spaces in indentationPylance" when translating Bitcounter on line 332.

**Comment by Chestshire** [comment]
Sorry, I clicked the wrong button. I thought it was "cancel comment", but it was actually `Close with comment`

**Comment by Chestshire** [comment]
> image
> 
> I am getting a "Inconsistent use of tabs and spaces in indentationPylance" when translating Bitcounter on line 332.

@Simonliuwaterloo @AidanCampbell738  I handled this in https://github.com/WatForm/org.alloytools.alloy/pull/23. Please take a look

**Review by Chestshire — APPROVED** [review]

**Pull request merged**